lib/init.php

```diff
@@ -57,3 +57,44 @@ function gutenberg_menu() {
 	);
 }
 add_action( 'admin_menu', 'gutenberg_menu', 9 );
+
+if ( ! function_exists( 'gutenberg_render_blocks_from_request' ) ) {
+	/**
+	 * Render blocks from a REST API request.
+	 *
+	 * @param mixed $request The current request.
+	 *
+	 * @return string
+	 */
+	function gutenberg_render_blocks_from_request( $request ) {
```

Comment:
Isn't this function kind of duplicating `do_blocks`. Can we just use it? What I also wonder is whether there's some block context we need to provide or anything. (Maybe be best to start without but I'm curious to see if there's any difference with the previous we're rendering right now)

Reply:
We defeinitely need to provide a global context - for example the query block needs to inherit some query. It's incoming. 

Yes we have so many functions :D I'll see about `do_blocks`.